### PR TITLE
ENH(merge?) adding function .to() for 2D

### PR DESCRIPTION
**Pull request by edouardoyallon**
- [x] Adding the source code for .to()
- [x] Editing the tests as proposed in #351 

The objective of this PR is to clearly separate the primitive used by a torch modules in order to prepare an integration nn.functional/nn.module-like

### CODE REVIEW

**Comment by edouardoyallon** [comment]
@janden actually, I'm not sure to see which test is affected by this change(for 2d), thus I'm pros merging!

**Comment by eickenberg** [comment]
This seems good to me.
Feel free to merge after you have added a test. If you do merge, please do it via squash and make sure the commit message starts with `ENH`.

If you feel super motivated, maybe you can add this also for 1D and 3D for API consistency? (Pretty please?)

File: kymatio/scattering2d/scattering2d.py [hunk 1]
```diff
@@ -107,26 +107,45 @@ def build(self):
         self.Psi = convert_filters(filters['psi'])
         self.Phi = convert_filters([filters['phi'][j] for j in range(self.J)])
 
-    def _type(self, _type):
+    def _apply(self, fn):
+        """
+            Mimicates the behavior of the function _apply() of a nn.Module()
```
Comment (eickenberg) [inline kymatio/scattering2d/scattering2d.py]: `s/Mimicates/Mimics` like everywhere. Maybe you can use `sed` lol

Comment (eickenberg) [reply kymatio/scattering2d/scattering2d.py]: https://stackoverflow.com/questions/11392478/how-to-replace-a-string-in-multiple-files-in-linux-command-line/20721292

File: kymatio/scattering2d/scattering2d.py [hunk 2]
```diff
@@ -107,26 +107,45 @@ def build(self):
...
-        self.pad.padding_module.type(_type)
+                    self.Psi[key][key2] = fn(item2)
+        self.Phi = [fn(v) for v in self.Phi]
+        self.pad.padding_module._apply(fn)
         return self
 
-    def cuda(self):
+    def cuda(self, device=None):
+        """
+            Mimicates the behavior of the function cuda() of a nn.Module()
```
Comment (eickenberg) [inline kymatio/scattering2d/scattering2d.py]: Mimics

File: kymatio/scattering2d/scattering2d.py (hunk 1 above)
Comment (edouardoyallon) [reply kymatio/scattering2d/scattering2d.py]: done, tks!!

**Pull request merged**